tests/integration/targets/docker_container/tasks/tests/ports.yml

```diff
@@ -303,149 +303,78 @@
+    publish_all_ports: "{{ test_case.publish_all_ports_value | default(omit) }}"
+    published_ports: "{{ test_case.published_ports_value | default(omit) }}"
     force_kill: yes
-  register: publish_all_ports_6
-
-- name: publish_all_ports (null -> true with published_ports)
-  docker_container:
-    image: "{{ docker_test_image_alpine }}"
-    command: '/bin/sh -c "sleep 10m"'
-    name: "{{ cname }}"
-    state: started
-    publish_all_ports: true
-    published_ports:
-      - "9001:9001"
-      - "9010-9050:9010-9050"
-    force_kill: yes
-  register: publish_all_ports_7
-
-- name: publish_all_ports (Idempotency with published_ports)
-  docker_container:
-    image: "{{ docker_test_image_alpine }}"
-    command: '/bin/sh -c "sleep 10m"'
-    name: "{{ cname }}"
-    state: started
-    publish_all_ports: true
-    published_ports:
-      - "9001:9001"
-      - "9010-9050:9010-9050"
-    force_kill: yes
-  register: publish_all_ports_8
-
-- name: publish_all_ports (true -> false with published_ports)
-  docker_container:
-    image: "{{ docker_test_image_alpine }}"
-    command: '/bin/sh -c "sleep 10m"'
-    name: "{{ cname }}"
-    state: started
-    publish_all_ports: false
-    published_ports:
-      - "9001:9001"
-      - "9010-9050:9010-9050"
-    force_kill: yes
-  register: publish_all_ports_9
-
-- name: publish_all_ports (Idempotency2 with published_ports)
-  docker_container:
-    image: "{{ docker_test_image_alpine }}"
-    command: '/bin/sh -c "sleep 10m"'
-    name: "{{ cname }}"
-    state: started
-    publish_all_ports: false
-    published_ports:
-      - "9001:9001"
-      - "9010-9050:9010-9050"
-    force_kill: yes
-  register: publish_all_ports_10
-
-- name: publish_all_ports (false -> null with published_ports)
-  docker_container:
-    image: "{{ docker_test_image_alpine }}"
-    command: '/bin/sh -c "sleep 10m"'
-    name: "{{ cname }}"
-    state: started
-    publish_all_ports: false
-    published_ports:
-      - "9001:9001"
-      - "9010-9050:9010-9050"
-    force_kill: yes
-  register: publish_all_ports_11
-
-- name: cleanup
-  docker_container:
-    name: "{{ cname }}"
-    state: absent
-    force_kill: yes
-  diff: no
+  register: publish_all_ports
+  loop_control:
+    loop_var: test_case
+  loop: "{{ publish_all_ports_test_cases }}"
 
 - assert:
     that:
-      - publish_all_ports_1 is changed
-      - publish_all_ports_2 is not changed
-      - publish_all_ports_3 is changed
-      - publish_all_ports_4 is not changed
-      - publish_all_ports_5 is not changed
-      - publish_all_ports_6 is changed
-      - publish_all_ports_7 is changed
-      - publish_all_ports_8 is not changed
-      - publish_all_ports_9 is changed
-      - publish_all_ports_10 is not changed
-      - publish_all_ports_11 is not changed
+      - publish_all_ports.results[index].changed == publish_all_ports_test_cases[index].changed
+  loop: "{{ range(0, publish_all_ports_test_cases | length) | list }}"
```

Comment:
For things like this, the `indexed_items` lookup is also pretty useful (https://docs.ansible.com/ansible/latest/collections/ansible/builtin/indexed_items_lookup.html#ansible-collections-ansible-builtin-indexed-items-lookup).

Reply:
Thanks, I updated it to the `loop` analouge.